python/pyarrow/types.pxi

```diff
@@ -817,6 +817,12 @@ def register_extension_type(ext_type):
     _python_extension_types_registry.append(_type)
 
 
+cdef _unregister_extension_type(bytes type_name):
+    # ARROW-8684: Due to a Cython (?) bug we cannot call "compat.tobytes"
```

Comment:
Calling `compat.tobytes` gives the same error???

Reply:
No, for some reason calling tobytes from within the atexit handler triggers the bug